src/plugins/histogram/index.js

```diff
@@ -0,0 +1,215 @@
+    this.getHistogramFile(this.options.src).then(function (res) {
+      _this2.setupHistogramElement();
+      if (_this2.histogramHolder) {
+        _this2.createHistogram(res);
+      }
+    });
+  };
+
+  /**
+   * Grabs the contents of the VTT file.
+   *
+   * @param url
+   * @returns {Promise}
+   */
+
+  HistogramPlugin.prototype.getHistogramFile = function getHistogramFile(url) {
+    let _this3 = this;
+
+    return new Promise(function (resolve) {
+      let req = new XMLHttpRequest();
+      req.data = {
+        resolve: resolve
+      };
+      req.addEventListener('load', _this3.histogramDataLoaded);
+      req.open('GET', url);
+      req.send();
+    });
+  };
+
+  /**
+   * Callback for loaded Histogram file.
+   */
+
+  HistogramPlugin.prototype.histogramDataLoaded = function histogramDataLoaded() {
+    this.data.resolve(JSON.parse(this.responseText));
+  };
+
+  HistogramPlugin.prototype.setupHistogramElement = function setupHistogramElement() {
+    let mouseDisplay = this.player.$('.vjs-mouse-display');
+    this.progressBar = this.player.$('.vjs-progress-control');
+    if (!this.progressBar) {
+      return;
+    }
+    let histogramHolder = this.player.$('.vjs-histogram-display') || document.createElement('div');
+    histogramHolder.setAttribute('class', 'vjs-histogram-display');
+    this.progressBar.appendChild(histogramHolder);
+    this.histogramHolder = histogramHolder;
+    if (mouseDisplay) {
+      mouseDisplay.classList.add('vjs-hidden');
+    }
+  };
+
+  /**
+   * Function to create the SVG path element
+   */
+  HistogramPlugin.prototype.createPath = function createPath(dataArray, containerWidth, containerHeight) {
+    // Calculate the x and y coordinates for each point
+    const stepX = containerWidth / (dataArray.length - 1);
+    const points = dataArray.map((value, index) => ({
+      x: index * stepX,
+      y: containerHeight - value * containerHeight
+    }));
+
+    // Create a smooth line path
+    const path = document.createElementNS('http://www.w3.org/2000/svg', 'path');
+    path.setAttribute('fill', 'lightblue');
+    // path.setAttribute('stroke', 'blue');
+
+    // Generate the smooth line path data
+    let d = `M ${points[0].x},${points[0].y}`;
+    for (let i = 0; i < points.length - 1; i++) {
+      const xc = (points[i].x + points[i + 1].x) / 2;
+      const yc = (points[i].y + points[i + 1].y) / 2;
+      d += ` Q ${points[i].x},${points[i].y} ${xc},${yc}`;
+    }
+    d += ` Q ${points[points.length - 1].x},${points[points.length - 1].y} ${points[points.length - 1].x},${points[points.length - 1].y}`;
+
+    // Close the path to fill the region under the line
+    d += ` L ${points[points.length - 1].x},${containerHeight} L ${points[0].x},${containerHeight} Z`;
+
+    path.setAttribute('d', d);
+
+    return path;
+  };
+
+  HistogramPlugin.prototype.createHistogram = function createHistogram(info) {
+    const data = info.data;
+    const svgWidth = 600;
+    const svgHeight = 20;
```

Comment:
should not these be dynamic based on the video element size?

Reply:
no, this is setting the SVG aspect ratio, rendered size is stretched to 100% width